esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,205 @@
+from esmvalcore.cmor.fix import Fix
+from esmvalcore.exceptions import RecipeError
+logger = logging.getLogger(__name__)
+
+
+class MOHCHadREM3GA705(Fix):
+    """General fix for MOHC-HadREM3-GA7-05."""
+
+    def fix_metadata(self, cubes):
+        """Fix time long_name, and latitude and longitude var_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('latitude').var_name = 'lat'
+            cube.coord('longitude').var_name = 'lon'
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class TimeLongName(Fix):
+    """Fixes for time coordinate."""
+
+    def fix_metadata(self, cubes):
+        """
+        Fix time long_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class CLMcomCCLM4817(Fix):
+    """Fixes for CLMcom-CCLM4-8-17."""
+
+    def fix_metadata(self, cubes):
+        """Fix calendars.
+
+        Set calendar to 'proleptic_gregorian' to avoid
+        concatenation issues between historical and
+        scenario runs.
+
+        Fix dtype value of coordinates and coordinate bounds.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            if cube.coord('time').units.calendar == 'standard':
+                cube.coord('time').units = Unit(
+                    'days since 1850-1-1 00:00:00',
+                    calendar='proleptic_gregorian'
+                )
+            for coord in cube.coords():
+                if coord.dtype in ['>f8', '>f4']:
+                    coord.points = coord.core_points().astype(
+                        np.float64, casting='same_kind')
+                    if coord.bounds is not None:
+                        coord.bounds = coord.core_bounds().astype(
+                            np.float64, casting='same_kind')
+        return cubes
+
+
+class AllVars(Fix):
+    """General CORDEX grid fix."""
+
+    def _check_grid_differences(self, old_coord, new_coord):
+        """Check differences between coords."""
+        diff = np.max(np.abs(old_coord.points - new_coord.points))
+        logger.debug(
+            "Maximum difference between original %s"
+            "points and standard %s domain points "
+            "for dataset %s and driver %s is: %s.",
+            new_coord.var_name,
+            self.extra_facets['domain'],
+            self.extra_facets['dataset'],
+            self.extra_facets['driver'],
+            str(diff)
+        )
+
+        if diff > 10e-4:
```

Comment:
I think we might want to have this threshold as a parameter. For different projections, the difference in real meters corresponding to the difference in projection units can be vastly different. Even when those projection units are "meters", which often then means "meters in the projection plane".

Reply:
I am not sure what you fully mean here. With @pepcos we tested this value with -11 and -44 domains and the datasets that passed the checks were quite decent. In any case I would leave it like that for now and if we see that with that threshold we are allowing datasets that are not good enough maybe we can find a way to tune it.

Reply:
Tuning it later sounds fine to me. Out of curiosity: If I understand it correctly, `-11` and `-44` refers to specific resolutions, but wouldn't you expect the projection and region to have a larger effect on this? Did you test with all CORDEX regions and all supported projections?